src/simulation_strategies/generalized_cliffords_b_simulation_strategy.py

```diff
@@ -0,0 +1,107 @@
+__author__ = "Tomasz Rybotycki"
+
+"""
+    This script contains the implementation of the Generalized Cliffords version B
+    algorithm. It differs from the original version  in two places:
+        -   In the k-th step, instead of computing k x k permanents, we compute a set of 
+            k-1 x k-1 permanents. From that, we calculate the required k x k permanents
+            in O(k) time each.
+        -   We permute the columns of the matrix at the beginning. We can then sample
+            from an easier distribution and obtain the same results.
+"""
+
+from .generalized_cliffords_simulation_strategy import GeneralizedCliffordsSimulationStrategy, BSPermanentCalculatorInterface
+from numpy import array, ndarray, int64
+from typing import List
+from numpy.random import choice
+from copy import deepcopy
+
+class GeneralizedCliffordsBSimulationStrategy(GeneralizedCliffordsSimulationStrategy):
+
+    def __init__(self, bs_permanent_calculator: BSPermanentCalculatorInterface) -> None:
+        self._current_input = []
+        self._working_input_state = None
+        super().__init__(bs_permanent_calculator)
```

Comment:
It probably doesn't really matter, but it's kinda "better" to call `super().__init__` at the start of the `__init__` method. It then "behaves" in a manner similar to other languages (e.g. C++).